### PR TITLE
Issue #424 passivation/activation and Annotated equality

### PR DESCRIPTION
**Pull request by jeanouii**
This small PR addresses the challenge of #424

### CODE REVIEW

File: impl/src/main/java/org/jboss/cdi/tck/tests/full/context/passivating/dependency/builtin/BuiltinBeanPassivationDependencyTest.java [hunk 1]
```diff
@@ -97,7 +97,11 @@ public void testInjectionPoint() throws IOException, ClassNotFoundException {
         assertEquals(inspectorCopy.getInjectionPoint().getQualifiers(), inspector.getInjectionPoint().getQualifiers());
         assertEquals(inspectorCopy.getInjectionPoint().getBean(), inspector.getInjectionPoint().getBean());
         assertEquals(inspectorCopy.getInjectionPoint().getMember(), inspector.getInjectionPoint().getMember());
-        assertEquals(inspectorCopy.getInjectionPoint().getAnnotated(), inspector.getInjectionPoint().getAnnotated());
+
+        // Annotated does not necessarly implement equals()/hashcode()
+        assertEquals(inspectorCopy.getInjectionPoint().getAnnotated().getBaseType(), inspector.getInjectionPoint().getAnnotated().getBaseType());
+        assertEquals(inspectorCopy.getInjectionPoint().getAnnotated().getAnnotations(), inspector.getInjectionPoint().getAnnotated().getAnnotations());
```
Comment (Ladicek) [inline impl/src/main/java/org/jboss/cdi/tck/tests/full/context/passivating/dependency/builtin/BuiltinBeanPassivationDependencyTest.java:103]: These are both good, but I think we should also (mainly?) assert that the underlying reflection objects are equal (which should work?). I know we already assert equality on `InjectionPoint.getMember()`, but I do think we should have both.

Something like

```java
assertEquals(
    unwrap(inspectorCopy.getInjectionPoint().getAnnotated()),
    unwrap(inspector.getInjectionPoint().getAnnotated())
);
```

where

```java
private static Object unwrap(Annotated annotated) {
    if (annotated instanceof AnnotatedMember) {
        return ((AnnotatedMember) annotated).getJavaMember();
    } else if (annotated instanceof AnnotatedParameter) {
        return ((AnnotatedParameter) annotated).getJavaParameter();
    } else if (annotated instanceof AnnotatedType) {
        return ((AnnotatedType) annotated).getJavaClass();
    } else {
        throw new UnsupportedOperationException("Unknown Annotated instance: " + annotated);
    }
}
```

WDYT?

Comment (jeanouii) [reply impl/src/main/java/org/jboss/cdi/tck/tests/full/context/passivating/dependency/builtin/BuiltinBeanPassivationDependencyTest.java:103]: Oh I see.
I'm happy to update the PR according to this. It looks more complete.

**Comment by jeanouii** [comment]
Weird, but yes I'll see if I can get ride of the last commit

**Comment by jeanouii** [comment]
Hopefully it's better. I extracted the method into the assert util class so it can be reused somewhere else if needed as your proposal was a bit more generic than what we needed

**Comment by jeanouii** [comment]
Added them back for completeness.

**Review by Ladicek — APPROVED** [review]

**Review by manovotn — APPROVED** [review]
Looks good, thanks!

**Pull request merged**